app/javascript/helpers/Axios.jsx

```diff
@@ -1,5 +1,35 @@
 import axios from 'axios';
 
+// Merges custom params with search params.
+function mergeConfigParams(params, configParams = { }) {
+  Object.entries(configParams).forEach(
+    (param) => {
+      const key = param.at(0);
+      const val = param.at(1);
+      if (val) {
+        params.set(key, val);
+      }
+    },
+  );
+
+  return params;
+}
```

Comment:
`mergeConfigParams(params: URLSearchParams, configParams: Object): URLSearchParams`
This function will be responsible for:
1. Taking in the search `params` extracted from the `window.location.search` and the `configParams` object given by the axios configuration used in a request `config.params` [MORE DETAILS](https://axios-http.com/docs/req_config).
2. Iterating over all axios configured params and add (merge) them to (with) the search `params` _if_ they have a truthy value since the config params allows `undefined|null|''` values to be passed for the JSON parser to remove.
-> The result will be a new list of params containing both of what extracted from the URL and what given to axios when setting the request therefore this function merges config params with search params.